lib/glob.js

```diff
@@ -16,7 +16,7 @@ function statList(fsm, list, callback) {
 
   var stats = [];
   var total = list.length;
-  for (var i = 0; i < CONC; ++i) {
+  for (var i = 0; i < list.length && i < CONC; ++i) {
```

Comment:
I'd consider a comparison with `Math.min()` more readable. Fans of premature optimization could save the `min()` result in a variable. ;-)